filebeat/crawler/registrar.go

```diff
@@ -74,7 +77,7 @@ func (r *Registrar) LoadState() {
 		defer existing.Close()
 		logp.Info("Loading registrar data from %s", r.registryFile)
 		decoder := json.NewDecoder(existing)
-		decoder.Decode(&r.State)
+		decoder.Decode(r.getFullState())
```

Comment:
When you need to write the entire `r.State` you’ll need to acquire the lock here because you can no longer use `r.getFullState()` if it’s returning a copy.